### PR TITLE
new functionalities in requirement doc

### PR DESCRIPTION
**Pull request by DanNiESh**
Add new reporting requirements for ESI Leap, including controller layer APIs and the updated manager service.

### CODE REVIEW

**Comment by tzumainn** [comment]
Hi! This looks like a good update to existing requirements. However, I don't see anything specific about reporting, which I should probably clarify.

Reporting is something that an esi-leap admin can see. It's a way for them to create, say, a spreadsheet of usage for a given time period - something that would tell them that project X leased node Y from timestamp A to timestamp B. Does that make sense?

**Comment by DanNiESh** [comment]
> Hi! This looks like a good update to existing requirements. However, I don't see anything specific about reporting, which I should probably clarify.
> 
> Reporting is something that an esi-leap admin can see. It's a way for them to create, say, a spreadsheet of usage for a given time period - something that would tell them that project X leased node Y from timestamp A to timestamp B. Does that make sense?

Yes. It makes sense. I'm updating it now.

**Comment by DanNiESh** [comment]
[Document ESI Leap Reporting Requirements](https://trello.com/c/Ea5Yj2P5/68-document-esi-leap-reporting-requirements)

File: docs/esi-leap-requirements.md [hunk 1]
```diff
@@ -39,9 +47,33 @@ Users can choose available nodes offered up and make contracts to lease nodes. A
...
 * expire offers: out-of-date offers, i.e, the current timestamp > offer's end_date, will be updated with an 'EXPIRED' status.
-* fulfill contracts: if a contract's start_date <= the current timestamp and is not expired, the manager service will fulfill the resources in the contracts and update the status of the contracts to 'FULFILLED'.
+* fulfill contracts: if a contract's start_date <= the current timestamp and is not expired, the manager service will fulfill the resources in the contracts and update the status of the contracts to 'FULFILLED'. 
 * expire contracts: same as 'expire offers', ESI Leap will expire contracts based on timestamp.
+* update offers: after the manager fulfills and expires contracts, it will update the relevant offers' status. The offers in a fulfilled contract should be unavailable to others. Likewise, when a contract expires, offers in the contract should be updated and be available again. 
+
+## Reporting Service
+This service reports to ESI Leap admins about the usage of resources given a period when an offer or a contract is created/updated. The reporting content should be like this:
+```
```
Comment (tzumainn) [inline docs/esi-leap-requirements.md:68]: Presumably there would be multiple entries?

File: docs/esi-leap-requirements.md [hunk 2]
```diff
@@ -39,9 +47,42 @@ Users can choose available nodes offered up and make contracts to lease nodes. A
...
 * expire offers: out-of-date offers, i.e, the current timestamp > offer's end_date, will be updated with an 'EXPIRED' status.
-* fulfill contracts: if a contract's start_date <= the current timestamp and is not expired, the manager service will fulfill the resources in the contracts and update the status of the contracts to 'FULFILLED'.
+* fulfill contracts: if a contract's start_date <= the current timestamp and is not expired, the manager service will fulfill the resources in the contracts and update the status of the contracts to 'FULFILLED'. 
 * expire contracts: same as 'expire offers', ESI Leap will expire contracts based on timestamp.
+* update offers: after the manager fulfills and expires contracts, it will update the relevant offers' status. The offers in a fulfilled contract should be unavailable to others. Likewise, when a contract expires, offers in the contract should be updated and be available again. 
+
+## Reporting Service
+This service reports to ESI Leap admin about the usage of nodes given a period. The admin enters a date range to get all contracts' information within that range. The results could be like this:
```
[the remark this answers is not in the document]
Comment (DanNiESh) [reply docs/esi-leap-requirements.md]: I see... I've changed it to 'Reporting API'.

**Pull request merged**